common/customizable_schema.go

```diff
@@ -30,6 +90,16 @@ func CustomizeSchemaPath(s map[string]*schema.Schema, path ...string) *Customiza
 	return &CustomizableSchema{Schema: sch, path: path}
 }
 
+func (s *CustomizableSchema) SchemaPath(path ...string) *CustomizableSchema {
+	sch := MustSchemaPath(s.GetSchemaMap(), path...)
+	return &CustomizableSchema{Schema: sch, path: path, context: s.context}
+}
+
+func (s *CustomizableSchema) GetSchemaMap(path ...string) map[string]*schema.Schema {
+	schemaMap := s.Schema.Elem.(*schema.Resource).Schema
```

Comment:
If `Elem` is nil or not a `*schema.Resource` we should panic but with a better error message.